python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -135,6 +137,16 @@ public PreciseIssue createIssue(SubscriptionContext ctx, Symbol symbol, Usage us
         issue.addQuickFix(quickFix);
       }
       return issue;
+    } else if (usage.kind().equals(Usage.Kind.ASSIGNMENT_LHS)) {
+      PreciseIssue issue = ctx.addIssue(usage.tree(), String.format(MESSAGE, symbol.name()));
+      AssignmentStatement assignmentStatement = ((AssignmentStatement) TreeUtils.firstAncestorOfKind(usage.tree(), Kind.ASSIGNMENT_STMT));
+      // This can only trigger if there is 1 lhs expression, the other cases are handled by isSequenceUnpacking
+      if (assignmentStatement != null) {
```

Comment:
From my understanding, this can happen when we have an annotated assignment, such as (in the tests):

```
def no_fn_type_annotation_with_assignment():
    value: str = "hello"  # Noncompliant
    return [int(value) for value in something()]
```

Maybe we could explicitly have a test showing no quick fix is offered in such case (and maybe we actually want to eventually cover that case as well - although you can technically trigger side effects through the evaluation of the type annotation...).

Reply:
As discussed, we will generate the quickfix even when there is a type annotation. We also allow the quick fix have this behaviour :  `value: str = "hello`  ->`"hello"`